AppManager/app_manager_server.py

```diff
@@ -203,8 +218,56 @@ def start_app(config):
     monit_interface.stop(watch)
     return BAD_PID
 
+  if 'log_size' in config.keys():
+    log_size = config['log_size']
+  else:
+    if config['app_name'] == APPSCALE_DASHBOARD_ID:
+      log_size = TEN_MB
+    else:
+      log_size = APP_LOG_SIZE
+
+  if not setup_logrotate(config['app_name'], watch, log_size):
+    logging.error("Error while setting up log rotation for application: {}".
+      format(config['app_name']))
+
   return 0
 
+def setup_logrotate(app_name, watch, log_size):
+  """ Creates a logrotate script for the logs that the given application
+      will create.
+
+  Args:
+    app_name: A string, the application ID.
+    watch: A string of the form 'app___<app_ID>'.
+    log_size: An integer, the size of logs that are kept per application server.
+      The size should be in bytes.
+  Returns:
+    True on success, False otherwise.
+  """
+  # Write application specific logrotation script.
+  app_logrotate_script = "{0}/appscale-{1}".format(LOGROTATE_CONFIG_DIR, app_name)
+
+  # Application logrotate script content.
+  contents = "/var/log/appscale/{0}*.log ".format(watch)
+  contents += "{"
+  contents += """
+  size {0}""".format(log_size)
+  contents += """
+  missingok
+  rotate 7
+  compress
+  delaycompress
+  notifempty
+  copytruncate
+}
+"""
```

Comment:
To me, this would be cleaner as either a single string declaration with a format statement or a template file.